paddle/fluid/operators/fused/mkldnn/multi_gru_mkldnn_op.cc

```diff
@@ -109,13 +109,8 @@ class MultiGRUHandler {
     const std::string unique_name = ctx.OutputName("Hidden");
     // Create memory key without Ti because weights, bias and h0 memories
     // do not depend on Ti size but primitive and input/output memory do
-    if (platform::MKLDNNDeviceContext::tls().get_cur_mkldnn_session_id() !=
-        platform::MKLDNNDeviceContextThreadLocals::kMKLDNNSessionID_Default) {
-      memory_key_ = CreateKey(unique_name, MKLDNNGetDataType<T>());
-    } else {
-      memory_key_ = CreateKey(unique_name, MKLDNNGetDataType<T>(), "-t:",
-                              platform::ThreadIDasStr());
-    }
+    memory_key_ = platform::ExtendKeyWithThreadInfoIfNeeded(
```

Comment:
Why for some operators the `CreateKey(...)` call is inside of the `ExtendedKeyWithThreadInfoIfNeeded(...)`, and for some operators it is not?

Reply:
CreateKey is not called inside ExtendKeyWithThreadInfoIfNeeded(). Create Key is either call on initialization list of MKLDNNHandlerT , or separatly in an op . This separate part is legacy code that need to be refactored so it does use MKLDNNHandlerT based handlers and then CreateKey will not be visible.

Reply:
OK, thank you!